.github/workflows/default.yaml

```diff
@@ -94,15 +94,29 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        os: [ ubuntu-22.04, macos-14 ]
+        os:
+          - ubuntu-22.04
+          - macos-14
+        compiler:
+          - { CC: gcc, CXX: g++ }
+          - { CC: clang, CXX: clang++ }
         layer:
-          - layer-00-default
-          - layer-01-minimal
-          - layer-02-maximus
-          - layer-04-noauth-everything
+          - { name: layer-00-default, nick: default }
+          - { name: layer-01-minimal, nick: minimal }
+          - { name: layer-02-maximus, nick: maximus }
+          - { name: layer-04-noauth-everything, nick: noauth }
```

Comment:
We have discussed removing this layer from these tests. If you want to remove this layer in _this_ PR, please commit the following suggestion:

```suggestion
```

One could argue that removing this layer is in this PR scope because this PR adds more tests (by adding a compiler dimension to the matric), and we are worried about having too many tests (in total).